cairo1-run/cairo1-programs/factorial.cairo

```diff
@@ -0,0 +1,14 @@
+use core::felt252;
+
+fn main() -> felt252 {
+    let n = 10;
+    let result = factorial(n);
+    result
+}
+
+fn factorial(n: felt252) -> felt252 {
+    match n {
+        0 => 1,
+        _ => n * factorial(n - 1),
+    }
+}
```

Comment:
We can move this programs to the top-level `cairo_programs` and make a `cairo_1_programs` folder inside it to keep all programs in the same directory